### ISSUE TITLE
ESQL: ROW operator, support parentheses in constant expressions (mv)

### ISSUE THREAD

**Issue by luigidellaquila** [elastic/elasticsearch#98396]
ROW operator accepts multi-value constants, consisting of square brackets and multiple values of the same type

```
ROW foo = [1, -12, 42]
```

The single values are treated as simple constants and do not accept constant expressions, eg. the following doesn't work

```
ROW foo = [1, -(-12), 42]
```

while the following (single value, no square brackets) works just fine

```
ROW foo = -(-12)
```

We should make this behavior consistent and accept parentheses and multiple unary operators also in square brackets

**Comment by bpintea** [comment]
Related, `ROW foo = [1 week, 2 weeks]` isn't accepted either, while `ROW foo = 1 week` is.

**Comment by astefan** [comment]
Good point. It is like that because the `constant` defines `integerValue UNQUOTED_IDENTIFIER` that handles `1 week` but the `OPENING_BRACKET .... (COMMA ....)* CLOSING_BRACKET` allows only numerics, booleans and strings.
